src/romein.cu

```diff
@@ -262,20 +178,35 @@ inline void launch_romein_kernel(int      nbaseline,
                                  InType*  d_in,
                                  OutType* d_out,
                                  cudaStream_t stream=0) {
-    //cout << "LAUNCH for " << nelement << endl;
-    dim3 block(8,1);
-    dim3 grid(nbatch*npol,1);
-    if( polmajor ) {
-        npol = 1;
-    } else {
-        block.y = npol;
-        grid.x = nbatch;
-    }
-    /*
-    cout << "  Block size is " << block.x << " by " << block.y << endl;
-    cout << "  Grid  size is " << grid.x << " by " << grid.y << endl;
-    */
     
+    cudaDeviceProp dev;
+    cudaError_t error;
+    error = cudaGetDeviceProperties(&dev, 0);
+     if(error != cudaSuccess)
+     {
+        printf("Error: %s\n", cudaGetErrorString(error));
+      }
+
+	
+    int blk_cnt ;
+    int block_x=maxsupport*maxsupport ;
+    int tile_y ;
+    if(block_x==1)tile_y=nbaseline;
+    else tile_y = dev.maxThreadsPerBlock/block_x;
```

Comment:
Need to consider resources in the context of all of bifrost -- allow user to restrict this.

Reply:
For optimal performance of any CUDA kernel a close to 100 % theoretical thread occupancy is ideal. It was in this view this was added, it is now rescaled to 50% of maximum threads so that the estimated theoretical occupancy is 50%. This will ensure that there are ample threads left with to be used. Although I do not consider not using it because it is only for one threadblock and threads within one thread block strictly do not run multiple kernels concurrently

Reply:
This makes it more generic now as it avoid the trial-and-error method of identifying the optimal thread block configuration for different GPUs.

Reply:
That may be true but it does not match the style that Bifrost is written in.

Reply:
It is now modified so that it uses only 50% of maximum threads within a thread block. Is that Ok or should I put specify the number of threads instead  ?

Reply:
Bifrost currently uses either a fixed block size or a block size that is scaled to one of the input parameters and is bounded by a fixed thread count.

Reply:
So in this case will it be fine to have the bound as 50% or 25 % or 15 % the maximum threads per block as our bounds ? Because the fixed thread count is certainly going to be some percent of the maximum threads right ? I'm just thinking so that the need to manually go for a trial and error could be potentially avoided for different GPUs, What do you think ?